blog_api/lib/blog_api_web/controllers/users_controller.ex

```diff
@@ -10,7 +10,26 @@ defmodule BlogApiWeb.UsersController do
          {:ok, token, _claims} <- Guardian.encode_and_sign(user) do
       conn
       |> put_status(:ok)
-      |> render("create.json", %{token: token})
+      |> render("login.json", %{token: token})
+    end
+  end
+
+  def login(conn, params) do
+    with :ok <- validate_params(params, "email"),
+         :ok <- validate_params(params, "password"),
+         {:ok, token} <- Guardian.authenticate(params) do
+      conn
+      |> put_status(:ok)
+      |> render("login.json", %{token: token})
+    end
+  end
+
+  defp validate_params(params, key) do
+    case Map.fetch(params, key) do
+      {:ok, nil} -> {:error, "\"#{key}\" is required"}
+      {:ok, ""} -> {:error, "\"#{key}\" can't be blank"}
+      :error -> {:error, "\"#{key}\" is required"}
+      {:ok, _value} -> :ok
```

Comment:
O que acha de você deixar esse `validate_params/2` como uma função a ser reutilizada em outros controllers? 

@AdrielBento , vc acha que vale utilizar Plug nesse cenário?

Reply:
@johnnytaira @AdrielBento como eu poderia transformar isso em algo reutilizável? Dei uma olhada nos plugs mas não consegui pensar em uma solução

Reply:
@johnnytaira @AdrielBento como eu poderia transformar isso em algo reutilizável? Dei uma olhada nos plugs mas não consegui pensar em uma solução

Reply:
@johnnytaira Daria para fazer, só não sei se daria para ser algo reutilizável. Vejo algumas opcoes:
1. Deixar no controller
2. Se nenhum endpoint mais for usar o validate_params, acho valido jogar ele para o `Guardian` module.
3. Usar uma lib especifica para tratar validação de schemas, para validar o payload. Ex: [ex_json_schema](https://github.com/jonasschmidt/ex_json_schema)